wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -1,20 +1,53 @@
 module Cardano.Wallet.WalletLayer.Types
     ( PassiveWalletLayer (..)
     , ActiveWalletLayer (..)
+    -- * Lenses
+    , pwlCreateWallet
+    , pwlGetWalletIds
+    , pwlGetWallet
+    , pwlUpdateWallet
+    , pwlDeleteWallet
+
+    , pwlCreateAccount
+    , pwlGetAccounts
+    , pwlGetAccount
+    , pwlUpdateAccount
+    , pwlDeleteAccount
+
+    , pwlGetAddresses
     ) where
 
-import           Universum ()
+import           Universum
 
-import           Cardano.Wallet.API.V1.Types (WalletId)
+import           Control.Lens (makeLenses)
+
+import           Cardano.Wallet.API.V1.Types (Account, AccountIndex, AccountUpdate, Address,
+                                              NewAccount, NewWallet, Wallet, WalletId, WalletUpdate)
 
 import           Cardano.Wallet.Kernel.Diffusion (WalletDiffusion (..))
 
 
 -- | The passive wallet (data) layer. See @PassiveWallet@.
 data PassiveWalletLayer m = PassiveWalletLayer
-    { pwlGetWalletIds       :: m [WalletId]
+    {
+    -- * wallets
+      _pwlCreateWallet  :: NewWallet -> m Wallet
+    , _pwlGetWalletIds  :: m [WalletId]
+    , _pwlGetWallet     :: WalletId -> m (Maybe Wallet)
+    , _pwlUpdateWallet  :: WalletId -> WalletUpdate -> m Wallet
+    , _pwlDeleteWallet  :: WalletId -> m Bool
+    -- * accounts
+    , _pwlCreateAccount :: WalletId -> NewAccount -> m Account
+    , _pwlGetAccounts   :: WalletId -> m [Account]
+    , _pwlGetAccount    :: WalletId -> AccountIndex -> m (Maybe Account)
+    , _pwlUpdateAccount :: WalletId -> AccountIndex -> AccountUpdate -> m Account
+    , _pwlDeleteAccount :: WalletId -> AccountIndex -> m Bool
+    -- * addresses
+    , _pwlGetAddresses  :: WalletId -> m [Address]
     }
 
+makeLenses ''PassiveWalletLayer
```

Comment:
This might be a good time to use `makeClassy` instead, which lets us use the `pwlCreateWallet` function on an `ActiveWallet` too. We would only need an `instance HasPassiveWallet (ActiveWalletLayer m)`.